src/app/hooks/use-profile-synchronizer.ts

```diff
@@ -0,0 +1,172 @@
+import { Profile } from "@arkecosystem/platform-sdk-profiles";
+import { useConfiguration, useEnvironmentContext } from "app/contexts";
+import { restoreProfileTestPassword } from "migrations";
+import { useEffect, useMemo, useRef } from "react";
+import { useLocation } from "react-router-dom";
+import { matchPath } from "react-router-dom";
+
+import { useNotifications } from "./notifications";
+import { useSynchronizer } from "./use-synchronizer";
+
+enum Intervals {
+	Short = 30000,
+	Medium = 60000,
+	Long = 120000,
+}
+
+const useProfileWatcher = () => {
+	const { env } = useEnvironmentContext();
+	const location = useLocation();
+	const pathname = (location as any).location?.pathname || location.pathname;
+	const match = useMemo(() => matchPath(pathname, { path: "/profiles/:profileId" }), [pathname]);
+	const profileId = (match?.params as any)?.profileId;
+	const allProfilesCount = env.profiles().count();
+
+	return useMemo(() => {
+		if (!profileId) return;
+		let response: Profile | undefined;
+
+		try {
+			response = env.profiles().findById(profileId);
+		} catch (e) {
+			// Not a valid profile id. Ignore.
+		}
+
+		return response;
+	}, [profileId, env, allProfilesCount]); // eslint-disable-line react-hooks/exhaustive-deps
+};
+
+const useProfileJobs = (profile?: Profile) => {
+	const { env } = useEnvironmentContext();
+	const { notifications } = useNotifications();
+
+	const walletsCount = profile?.wallets().count();
+	return useMemo(() => {
+		if (!profile) return [];
+
+		const syncDelegates = {
+			callback: () => env.delegates().syncAll(),
+			interval: Intervals.Long,
+		};
+
+		const syncFees = {
+			callback: () => env.fees().syncAll(),
+			interval: Intervals.Medium,
+		};
+
+		const syncExchangeRates = {
+			callback: () => {
+				const currencies = Object.keys(env.coins().all());
+				return Promise.all(currencies.map((currency) => env.exchangeRates().syncAll(profile, currency)));
+			},
+			interval: Intervals.Long,
+		};
+
+		const syncWallets = {
+			callback: () => env.wallets().syncByProfile(profile),
+			interval: Intervals.Short,
+		};
+
+		const syncNotifications = {
+			callback: () => notifications.notifyReceivedTransactions({ profile }),
+			interval: Intervals.Short,
+		};
+
+		return [syncWallets, syncFees, syncDelegates, syncExchangeRates, syncNotifications];
+	}, [env, profile, walletsCount, notifications]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Why is eslint disabled for this line?

Reply:
@faustbrian same reason as in `Wallets`. Needed to update profile jobs when wallets are added with new networks, coins, etc, but not using the variable inside the hook.